atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -112,6 +113,14 @@ def run(self):
             self.log.info(str(e))
             self.log.info('Canceling the compose %s', e.compose_id)
             self.odcs_client.cancel_compose(e.compose_id)
+
+            # Cancel also composes which were created by the build
+            preserve_composes = set(self.compose_ids).union(set(self.parent_compose_ids))
```

Comment:
`.union` doesn't need `set` just `iterable`
```suggestion
            preserve_composes = set(self.compose_ids).union(self.parent_compose_ids)
```